modules/lang-painless/src/main/java/org/elasticsearch/painless/ir/CatchNode.java

```diff
@@ -96,15 +92,14 @@ protected void write(ClassWriter classWriter, MethodWriter methodWriter, WriteSc
         methodWriter.visitVarInsn(variable.getAsmType().getOpcode(Opcodes.ISTORE), variable.getSlot());
 
         if (blockNode != null) {
-            blockNode.continueLabel = continueLabel;
-            blockNode.breakLabel = breakLabel;
-            blockNode.write(classWriter, methodWriter, writeScope);
+            blockNode.write(writeScope.newTryScope(null, null, null));
         }
 
-        methodWriter.visitTryCatchBlock(begin, end, jump, variable.getAsmType().getInternalName());
+        methodWriter.visitTryCatchBlock(
+                writeScope.getTryBeginLabel(), writeScope.getTryEndLabel(), jump, variable.getAsmType().getInternalName());
```

Comment:
Why does method stuff things into the write scope to just pull them out?  Maybe `visitTryCatchBlock` can take in a `WriteScope`?

Reply:
This is the actual method to write the ASM instruction for a try/catch block and requires all these labels.